hack/src/check_releases.rs

```diff
@@ -0,0 +1,40 @@
+use cincinnati::plugins::internal::release_scrape_dockerv2::plugin;
+use cincinnati::plugins::internal::release_scrape_dockerv2::registry;
+
+use anyhow::Result as Fallible;
+use anyhow::{bail, Context};
+use semver::Version;
+use std::collections::HashSet;
+
+pub async fn run(found_versions: HashSet<Version>) -> Fallible<()> {
+  let settings = plugin::ReleaseScrapeDockerv2Settings::default();
+  let cache = registry::cache::new();
+  let registry = registry::Registry::try_from_str(&settings.registry)
+    .context(format!("Parsing {} as Registry", &settings.registry))?;
+
+  println!("Scraping Quay registry");
+  let releases = registry::fetch_releases(
+    &registry,
+    &settings.repository,
+    settings.username.as_ref().map(String::as_ref),
+    settings.password.as_ref().map(String::as_ref),
+    cache,
+    &settings.manifestref_key,
+    settings.fetch_concurrency,
+  )
+  .await
+  .context("failed to fetch all release metadata")?;
+
+  println!("Verifying all releases are uploaded");
+  for v in found_versions.iter() {
+    if releases
+      .iter()
+      .find(|&r| r.metadata.version == *v)
+      .is_none()
+    {
+      bail!("Version {} is not found in scraped images", v)
+    }
+  }
```

Comment:
I think we can be more efficient here by comparing two sets. (Typing in the github comment window here without actually verifying the code, please beware :wink:)

```suggestion
  .context("failed to fetch all release metadata")?
  .into_iter()
  .map(|r| r.metadata.version)
  .collect::<HashSet<Version>>();

  println!("Verifying all releases are uploaded");
  // assume we've assigned the above HashSet to `released_versions`
  let missing_versions = found_versions - released_versions;
  if !missing_versions.is_empty() {
    bail!("Missing the following versions in scraped images: {:?}", missing_versions)
  }
```

Reply:
Good idea, implemented this using `HashSet::Difference`